
syntax = "proto3";

message StoreData
{
    string filename = 1;
}

message GetData 
{
    string filename = 1; 
}

message Fragment_Status_Request
{
    string fragment_name = 1; 
}

message Fragment_Status_Response
{
    string fragment_name = 1;
    bool is_present = 2;
    string node_id = 3;
}

enum request_type
{
    FRAGMENT_STATUS_REQ = 0; 
    FRAGMENT_DATA_REQ = 1;
    STORE_FRAGMENT_DATA_REQ = 2; 
}

message header
{
    request_type request_type = 1; 
}